File: Data/Entities/PlaylistItem.cs
﻿using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace yt_logger.Data.Entities
{
    public class PlaylistItem : BaseEntity
    {
        [Key]
        public string RefId { get; set; }
        public string Title { get; set; }
        public string ImgUrl { get; set; } //save locally, cause if deleted video, image deletes too
        public string VideoOwnerChannelTitle { get; set; }
        public string VideoOwnerChannelId { get; set; }
        public DateTime VideoPublishedAt { get; set; }
        public bool IsDeleted { get; set; }
    }
}

File: Ui/src/model/playlistItem.ts
export interface Playlist {
  title: string;
  refId: string;
  itemCount: number;
  lastLogged?: string;
  imgUrl: string;
}

export interface PlaylistItem {
  title: string;
  imgUrl: string;
  videoOwnerChannelTitle: string;
  videoOwnerChannelId: string;
  videoPublishedAt: string;
  refId: string;
  isDeleted: boolean;
  playlistId: number;
}

export interface Log {
  id: number;
  timeStamp: string;
  refId: string;
  added: PlaylistItem[];
  deleted: PlaylistItem[];
}

export interface PagedResult {
  result: Log[];
  pageCount: number;
}
